### PR TITLE
modified with new version supports for k8 1.26

### PR DESCRIPTION
**Pull request by miyurudhammage-eroad**

### CODE REVIEW

**Comment by jpeaks-eroad** [comment]
Logo
**Checkmarx One – Scan Summary & Details** – [023572e2\-61cb\-44f8\-b2f6\-8be0d1ae8281](https://anz.ast.checkmarx.net/projects/72341a0d-af92-44ff-b4bc-132183d26b87/scans?id=023572e2-61cb-44f8-b2f6-8be0d1ae8281&branch=kube-janitor-v3)
### New Issues
Severity|Issue|Source File / Package|Checkmarx Insight
---|---|---|---
HIGH|[CVE-2023-37920](https://devhub.checkmarx.com/cve-details/CVE-2023-37920)|Python\-certifi\-2023\.5\.7|[Vulnerable Package](https://anz.ast.checkmarx.net/sca/#/projects/72341a0d-af92-44ff-b4bc-132183d26b87/reports/023572e2-61cb-44f8-b2f6-8be0d1ae8281/vulnerabilities/CVE-2023-37920%3APython-certifi-2023.5.7/vulnerabilityDetails)
HIGH|[CVE-2023-43804](https://devhub.checkmarx.com/cve-details/CVE-2023-43804)|Python\-urllib3\-2\.0\.3|[Vulnerable Package](https://anz.ast.checkmarx.net/sca/#/projects/72341a0d-af92-44ff-b4bc-132183d26b87/reports/023572e2-61cb-44f8-b2f6-8be0d1ae8281/vulnerabilities/CVE-2023-43804%3APython-urllib3-2.0.3/vulnerabilityDetails)
HIGH|Missing User Instruction|/Dockerfile: [1](https://github.com/eroad/kube-janitor/blob/kube-janitor-v3//Dockerfile#L1)|A user should be specified in the dockerfile, otherwise the image will run as root
MEDIUM|[CVE-2023-45803](https://devhub.checkmarx.com/cve-details/CVE-2023-45803)|Python\-urllib3\-2\.0\.3|[Vulnerable Package](https://anz.ast.checkmarx.net/sca/#/projects/72341a0d-af92-44ff-b4bc-132183d26b87/reports/023572e2-61cb-44f8-b2f6-8be0d1ae8281/vulnerabilities/CVE-2023-45803%3APython-urllib3-2.0.3/vulnerabilityDetails)
MEDIUM|Pip install Keeping Cached Packages|/Dockerfile: [5](https://github.com/eroad/kube-janitor/blob/kube-janitor-v3//Dockerfile#L5)|When installing packages with pip, the '--no-cache-dir' flag should be set to make Docker images smaller
MEDIUM|Unpinned Package Version in Pip Install|/Dockerfile: [5](https://github.com/eroad/kube-janitor/blob/kube-janitor-v3//Dockerfile#L5)|Package version pinning reduces the range of versions that can be installed, reducing the chances of failure due to unanticipated changes
LOW|Healthcheck Instruction Missing|/Dockerfile: [1](https://github.com/eroad/kube-janitor/blob/kube-janitor-v3//Dockerfile#L1)|Ensure that HEALTHCHECK is being used. The HEALTHCHECK instruction tells Docker how to test a container to check that it is still working
LOW|Multiple RUN, ADD, COPY, Instructions Listed|/Dockerfile: [7](https://github.com/eroad/kube-janitor/blob/kube-janitor-v3//Dockerfile#L7)|Multiple commands (RUN, COPY, ADD) should be grouped in order to reduce the number of layers.
LOW|Using Unnamed Build Stages|/Dockerfile: [18](https://github.com/eroad/kube-janitor/blob/kube-janitor-v3//Dockerfile#L18)| This query is used to ensure that build stages are named. This way even if the Dockerfile is re-ordered, the COPY instruction doesn’t break.

### Fixed Issues
Severity|Issue|Source File / Package
---|---|---
HIGH|[CVE-2020-14343](https://devhub.checkmarx.com/cve-details/CVE-2020-14343)|Poetry\-pyyaml\-5\.3\.1
HIGH|[CVE-2020-29651](https://devhub.checkmarx.com/cve-details/CVE-2020-29651)|Poetry\-py\-1\.9\.0
HIGH|[CVE-2021-33503](https://devhub.checkmarx.com/cve-details/CVE-2021-33503)|Poetry\-urllib3\-1\.22
HIGH|[CVE-2022-23491](https://devhub.checkmarx.com/cve-details/CVE-2022-23491)|Poetry\-certifi\-2020\.6\.20
HIGH|[CVE-2022-42969](https://devhub.checkmarx.com/cve-details/CVE-2022-42969)|Poetry\-py\-1\.9\.0
HIGH|Missing User Instruction|/Dockerfile: [1](https://github.com/eroad/kube-janitor/blob/kube-janitor-v3//Dockerfile#L1)
MEDIUM|[CVE-2020-26137](https://devhub.checkmarx.com/cve-details/CVE-2020-26137)|Poetry\-urllib3\-1\.22
MEDIUM|Pip install Keeping Cached Packages|/Dockerfile: [5](https://github.com/eroad/kube-janitor/blob/kube-janitor-v3//Dockerfile#L5)
MEDIUM|Unpinned Package Version in Pip Install|/Dockerfile: [5](https://github.com/eroad/kube-janitor/blob/kube-janitor-v3//Dockerfile#L5)
LOW|Healthcheck Instruction Missing|/Dockerfile: [1](https://github.com/eroad/kube-janitor/blob/kube-janitor-v3//Dockerfile#L1)
LOW|Multiple RUN, ADD, COPY, Instructions Listed|/Dockerfile: [7](https://github.com/eroad/kube-janitor/blob/kube-janitor-v3//Dockerfile#L7)
LOW|Using Unnamed Build Stages|/Dockerfile: [18](https://github.com/eroad/kube-janitor/blob/kube-janitor-v3//Dockerfile#L18)

File: unsupported/helm/kube-janitor-0.3.1.tgz
Comment (eroadrdrdog) [inline unsupported/helm/kube-janitor-0.3.1.tgz:1]: The folder in this zip is named "kube-janitor 2". I think it would probably need to be "kube-janitor".

**Review by ilyakotov-eroad — APPROVED** [review]
LGTM, let's try this

File: unsupported/helm/kube-janitor-0.3.1.tgz
Comment (miyurudhammage-eroad) [reply unsupported/helm/kube-janitor-0.3.1.tgz:1]: I couldn't find it 
Screenshot 2023-11-27 at 4 10 45 PM
I get it like this when extracted the 0.3.1

Comment (eroadrdrdog) [reply unsupported/helm/kube-janitor-0.3.1.tgz:1]: Sorry - my mistake. I blame finder :)

**Review by eroadrdrdog — APPROVED** [review]

**Pull request merged**